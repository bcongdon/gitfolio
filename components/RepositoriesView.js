import React from "react";
import { List, Container, Content } from "native-base";
import RepositoryListItem from "./RepositoryListItem";
import RepoModal from "./RepoModal";
import Modal from "react-native-modal";
import { View } from "react-native";
/*
  RepositoriesView is the component that displays a users list of public repositories
 */
export default class RepositoriesView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
      selectedRepo: -1
    };
  }

  /**
   * Renders a single repo ListItem
   * @param  {object} repo The repo object
   */
  renderRepo(repo, idx) {
    let key = `${repo.name}|${repo.owner.login}`;

    return (
      <RepositoryListItem
        key={key}
        name={repo.name}
        description={repo.description}
        language={repo.primaryLanguage ? repo.primaryLanguage.name : 0}
        stars={repo.stargazers ? repo.stargazers.totalCount : 0}
        viewerHasStarred={repo.viewerHasStarred}
        pushedAt={repo.pushedAt}
        url={repo.url}
        owner={repo.owner}
        onSetStarred={newVal =>
          this.props.setStarred(repo.owner.login, repo.name, newVal)
        }
        onPress={() => this.onRepoClick(idx)}
      />
    );
  }

  /**
   * Opens the repository modal for the specified repo
   * @param  {number} repoIdx Index of the clicked repo
   */
  onRepoClick(repoIdx) {
    this.setState({
      modalVisible: true,
      selectedRepo: repoIdx
    });
  }

  /**
   * Renders a repository modal for the provided repo
   * @param  {object} repo The object representing the repo for the modal
   */
  renderRepoModal(repo) {
    if (!repo) {
      return <View />;
    }

    return (
      <RepoModal
        name={repo.name}
        description={repo.description}
        language={repo.primaryLanguage ? repo.primaryLanguage.name : 0}
        stars={repo.stargazers ? repo.stargazers.totalCount : 0}
        viewerHasStarred={repo.viewerHasStarred}
        pushedAt={repo.pushedAt}
        url={repo.url}
        owner={repo.owner}
        onSetStarred={newVal =>
          this.props.setStarred(repo.owner.login, repo.name, newVal)
        }
      />
    );
  }

  /**
   * Renders the list of repositories
   */
  render() {
    return (
      <Container>
        <Content>
          <List>
            {this.props.repos.map((item, idx) => this.renderRepo(item, idx))}
          </List>
        </Content>
        <Modal
          isVisible={this.state.modalVisible}
          onBackdropPress={() => this.setState({ modalVisible: false })}
        >
          {this.renderRepoModal(this.props.repos[this.state.selectedRepo])}
        </Modal>
      </Container>
    );
  }
}
