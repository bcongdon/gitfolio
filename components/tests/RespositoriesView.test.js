import RepositoriesView from "../RepositoriesView";
import React from "react";

import renderer from "react-test-renderer";

it("renders without crashing", () => {
  let component = <RepositoriesView repos={[]} />;

  const rendered = renderer.create(component).toJSON();
  expect(rendered).toBeTruthy();
});

it("should pass snapshot test", () => {
  let component = (
    <RepositoriesView
      repos={[
        { name: "test repo", description: "foo bar", owner: { login: "foo" } }
      ]}
    />
  );

  const rendered = renderer.create(component).toJSON();
  expect(rendered).toMatchSnapshot();
});
