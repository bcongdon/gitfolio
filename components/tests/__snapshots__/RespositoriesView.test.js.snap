// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass snapshot test 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            button={true}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "center",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                test repo
                

                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  note={true}
                  style={
                    Object {
                      "color": "#a7a7a7",
                      "fontFamily": "System",
                      "fontSize": 14,
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  foo bar
                </Text>
                


                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                  />
                   
                  a few seconds ago
                </Text>
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "alignSelf": "center",
                  "flex": 0.28,
                  "paddingBottom": null,
                  "paddingLeft": null,
                  "paddingRight": null,
                  "paddingTop": null,
                }
              }
              virtual={undefined}
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": null,
                    "backgroundColor": null,
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 5,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "elevation": 0,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                    "shadowColor": null,
                    "shadowOffset": null,
                    "shadowOpacity": null,
                    "shadowRadius": null,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
      testID={undefined}
    />
    <View
      collapsable={undefined}
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "margin": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      swipeThreshold={100}
    >
      <View />
    </View>
  </Modal>
</View>
`;
